great_expectations/util.py

```diff
@@ -1213,15 +1214,80 @@ def filter_properties_dict(
     return properties
 
 
+@overload
 def deep_filter_properties_iterable(
-    properties: Optional[Any] = None,
+    properties: dict,
+    keep_fields: Optional[Set[str]] = ...,
+    delete_fields: Optional[Set[str]] = ...,
+    clean_nulls: bool = ...,
+    clean_falsy: bool = ...,
+    keep_falsy_numerics: bool = ...,
+    inplace: bool = ...,
+) -> dict:
+    ...
+
+
+@overload
+def deep_filter_properties_iterable(
+    properties: list,
+    keep_fields: Optional[Set[str]] = ...,
+    delete_fields: Optional[Set[str]] = ...,
+    clean_nulls: bool = ...,
+    clean_falsy: bool = ...,
+    keep_falsy_numerics: bool = ...,
+    inplace: bool = ...,
+) -> list:
+    ...
+
+
+@overload
+def deep_filter_properties_iterable(
+    properties: set,
+    keep_fields: Optional[Set[str]] = ...,
+    delete_fields: Optional[Set[str]] = ...,
+    clean_nulls: bool = ...,
+    clean_falsy: bool = ...,
+    keep_falsy_numerics: bool = ...,
+    inplace: bool = ...,
+) -> set:
+    ...
+
+
+@overload
+def deep_filter_properties_iterable(
+    properties: tuple,
+    keep_fields: Optional[Set[str]] = ...,
+    delete_fields: Optional[Set[str]] = ...,
+    clean_nulls: bool = ...,
+    clean_falsy: bool = ...,
+    keep_falsy_numerics: bool = ...,
+    inplace: bool = ...,
+) -> tuple:
+    ...
+
+
+@overload
+def deep_filter_properties_iterable(
+    properties: None,
+    keep_fields: Optional[Set[str]] = ...,
+    delete_fields: Optional[Set[str]] = ...,
+    clean_nulls: bool = ...,
+    clean_falsy: bool = ...,
+    keep_falsy_numerics: bool = ...,
+    inplace: bool = ...,
+) -> None:
+    ...
```

Comment:
These overloads are purely for the typing system. They do not alter runtime behavior at all.

Reply:
ah very interesting, thanks for the explanation - would it be possible to use a generic here? Takes in an object of type `T` and returns `T`?

Reply:
@cdkini Yes, great point. Should have done that from the beginning. Also Should reduce the number of overloads from 4 -> 2

Reply:
I expected this to work but keep getting `assignment` errors in the body of the function when trying to use generics.
https://github.com/great-expectations/great_expectations/pull/6099#pullrequestreview-1122468188

Going to revert this for now, but I would prefer to do these using generics.